Chained methods inheritances hide virtual function

Compiling your demo with GCC 14.2 and `-std=c++23 -Wall -Wextra -Werror` raises the error:

```
erased/include/erased/erased.h:90:15: error: ‘constexpr R erased::details::concrete::concrete_method<Base, Type, erased::Signature<Method, R(Args ...) const>, Others ...>::invoker(Method, Args ...) const [with Base = erased::basic_erased<32, Draw, erased::Move, erased::Copy>::Base; Type = Rectangle; Method = erased::Move; R = void; Args = {}; Others = {erased::Signature<erased::Copy, void() const>}]’ was hidden [-Werror=overloaded-virtual=]
   90 |   constexpr R invoker(Method, Args... args) const override {
      |               ^~~~~~~
erased/include/erased/erased.h:90:15: note:   by ‘constexpr R erased::details::concrete::concrete_method<Base, Type, erased::Signature<Method, R(Args ...) const>, Others ...>::invoker(Method, Args ...) const [with Base = erased::basic_erased<32, Draw, erased::Move, erased::Copy>::Base; Type = Rectangle; Method = Draw; R = void; Args = {int&}; Others = {erased::Signature<erased::Move, void() const>, erased::Signature<erased::Copy, void() const>}]’
```

I only change the Drawable type to be movable and copiable:
```cpp
using Drawable = erased::erased<Draw, erased::Move, erased::Copy>;
```

It seams virtual invoker method signatures are different between Methods.
Are you trying a composition pattern instead of derivation?

\~\~\~
Your project is very interesting! It gives me a lot of ideas to replaces template derivation for non-virtual types :)!

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@ It is a C++ type-erasure implementation developed with performance and ease of u
 
 It is heavily inspired by [AnyAny](https://github.com/kelbon/AnyAny).
 
-The [tested compilers](https://godbolt.org/z/ss8PE6zc3) are:
+The [tested compilers](https://godbolt.org/z/dWe3oPceW) are:
 1. MSVC (19.32 and above): `constexpr` does not work since MSVC does not yet handle `virtual` functions in a constexpr context.
 2. Clang (19.1 and above)
 3. GCC (14.1 and above)
@@ -80,3 +80,6 @@ For example, if you want to have a copyable and movable Drawable, you can do
 // Draw behavior
 using Drawable = erased::erased<Draw, erased::Move, erased::Copy>;
 ```
+## Thanks
+Here is the list of people who help me to develop and test this library:
+1. [Théo Devaucoup](https://github.com/theo-dep)

--- a/benchmarks/CMakeLists.txt
+++ b/benchmarks/CMakeLists.txt
@@ -8,5 +8,5 @@ set(BENCHMARK_ENABLE_TESTING OFF)
 FetchContent_MakeAvailable(googlebench)
 
 add_executable(Benchmarks benchmarks.cpp)
-target_link_libraries(Benchmarks PRIVATE erased::erased benchmark::benchmark)
+target_link_libraries(Benchmarks PRIVATE erased::erased erased::warnings benchmark::benchmark)
 

--- a/erased/CMakeLists.txt
+++ b/erased/CMakeLists.txt
@@ -16,6 +16,7 @@ set(gcc_cxx "$<COMPILE_LANG_AND_ID:CXX,GNU>")
 set(msvc_cxx "$<COMPILE_LANG_AND_ID:CXX,MSVC>")
 
 add_library(erased_sanitizer INTERFACE)
+
 target_compile_options(erased_sanitizer INTERFACE
     $<${clang_cxx}:-fsanitize=address;-fsanitize=undefined>
     $<${msvc_cxx}:/fsanitize=address;/Zc:preprocessor>
@@ -29,6 +30,14 @@ target_compile_options(erased_sanitizer INTERFACE
     $<${msvc_cxx}:-D_DISABLE_VECTOR_ANNOTATION;-D_DISABLE_STRING_ANNOTATION>
 )
 
+add_library(erased_warnings INTERFACE)
+
+target_compile_options(erased_warnings INTERFACE
+    $<${clang_cxx}:-Wall -Wextra -Werror>
+    $<${gcc_cxx}:-Wall -Wextra -Werror>
+)
+
 add_library(erased::erased ALIAS erased)
 add_library(erased::sanitizer ALIAS erased_sanitizer)
+add_library(erased::warnings ALIAS erased_warnings)
 

--- a/erased/include/erased/erased.h
+++ b/erased/include/erased/erased.h
@@ -42,10 +42,10 @@ template <typename Method>
 using CreateSignature = Signature<Method, MethodPtr<Method>>;
 
 struct Copy {
-  static void invoker(const auto &) {}
+  static constexpr void invoker(const auto &) {}
 };
 struct Move {
-  static void invoker(const auto &) {}
+  static constexpr void invoker(const auto &) {}
 };
 
 namespace details::base {
@@ -87,6 +87,8 @@ template <typename Base, typename Type, typename Method, typename R,
 struct concrete_method<Base, Type, Signature<Method, R(Args...) const>,
                        Others...> : concrete_method<Base, Type, Others...> {
   using concrete_method<Base, Type, Others...>::concrete_method;
+  using concrete_method<Base, Type, Others...>::invoker;
+
   constexpr R invoker(Method, Args... args) const override {
     return Method::invoker(this->m_object, fwd(args)...);
   }
@@ -97,6 +99,8 @@ template <typename Base, typename Type, typename Method, typename R,
 struct concrete_method<Base, Type, Signature<Method, R(Args...)>, Others...>
     : concrete_method<Base, Type, Others...> {
   using concrete_method<Base, Type, Others...>::concrete_method;
+  using concrete_method<Base, Type, Others...>::invoker;
+
   constexpr R invoker(Method, Args... args) override {
     return Method::invoker(this->m_object, fwd(args)...);
   }

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -6,4 +6,4 @@ FetchContent_Declare(googletest
 FetchContent_MakeAvailable(googletest)
 
 add_executable(Tests tests.cpp)
-target_link_libraries(Tests PRIVATE erased::erased erased::sanitizer gtest_main gtest)
+target_link_libraries(Tests PRIVATE erased::erased erased::sanitizer erased::warnings gtest_main gtest)

